packages/ra-core/src/auth/useLogoutIfAccessDenied.ts

```diff
@@ -43,6 +43,15 @@ const useLogoutIfAccessDenied = (): LogoutIfAccessDenied => {
     const logout = useLogout();
     const notify = useNotify();
     const navigate = useNavigate();
+
+    const handleRedirect = (url: string) => {
+        if (url.startsWith('http')) {
+            window.location.href = url;
+        } else {
+            navigate(url);
+        }
+    };
```

Comment:
Which means you probably need to memoize this callback to avoid recreating a new one on each rerender